homeassistant/components/velbus/__init__.py

```diff
@@ -1,81 +1,44 @@
 """Support for Velbus devices."""
 import logging
 
-import velbus
+from velbusaio.controller import Velbus
 import voluptuous as vol
 
-from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
-from homeassistant.const import CONF_ADDRESS, CONF_NAME, CONF_PORT
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, CONF_PORT
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import ConfigEntryNotReady
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
 
 from .const import CONF_MEMO_TEXT, DOMAIN, SERVICE_SET_MEMO_TEXT
 
 _LOGGER = logging.getLogger(__name__)
 
-VELBUS_MESSAGE = "velbus.message"
-
 CONFIG_SCHEMA = vol.Schema(
     {DOMAIN: vol.Schema({vol.Required(CONF_PORT): cv.string})}, extra=vol.ALLOW_EXTRA
 )
 
 PLATFORMS = ["switch", "sensor", "binary_sensor", "cover", "climate", "light"]
 
 
-async def async_setup(hass, config):
-    """Set up the Velbus platform."""
-    # Import from the configuration file if needed
-    if DOMAIN not in config:
-        return True
-    port = config[DOMAIN].get(CONF_PORT)
-    data = {}
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Establish connection with velbus."""
+    hass.data.setdefault(DOMAIN, {})
 
-    if port:
-        data = {CONF_PORT: port, CONF_NAME: "Velbus import"}
-    hass.async_create_task(
-        hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_IMPORT}, data=data
-        )
-    )
+    controller = Velbus(entry.data[CONF_PORT])
+    hass.data[DOMAIN][entry.entry_id] = controller
+    await controller.connect()
 
-    return True
+    for platform in PLATFORMS:
+        hass.add_job(hass.config_entries.async_forward_entry_setup(entry, platform))
```

Comment:
Use this instead:
https://github.com/home-assistant/core/blob/bc417162cf264a5561555f3ce74ad7bdb9bbca69/homeassistant/config_entries.py#L1049-L1052